test/smoke/smoke_test.js

```diff
@@ -0,0 +1,102 @@
+const path = require('path');
+const childProcess = require('child_process');
+const util = require('util');
+const fs = require('fs');
+
+// Catch errors from failing promises
+process.on('unhandledRejection', function (error) {
+  console.error(error && error.stack);
+  process.exit(3);
+});
+
+const configsDir = path.normalize(path.join(__dirname, '../configs'));
+
+const processEnv = Object.assign({}, process.env);
+const ciMode = process.argv.includes('ci');
+if (ciMode) {
+  processEnv.BACKSTOP_CI_MODE = '1';
+}
+
+function sleep (ms) {
+  return new Promise(resolve => setTimeout(resolve, ms));
+}
+
+async function runSmokeTest (description, commands) {
+  const child = childProcess.spawn(commands[0], commands.slice(1),
+    { cwd: configsDir, env: processEnv }
+  );
+
+  const requiredStdout = [
+    'Element not found for capturing: .monkey',
+    'report | 0 Failed'
+  ];
+
+  const allowedStdErr = [
+    'Possible EventEmitter memory leak detected.'
+  ];
+
+  let missingStdOut = [...requiredStdout];
+
+  child.stdout.on('data', function (buf) {
+    const line = String(buf);
+    util.print(line);
+    const key = line.trim();
+    missingStdOut = missingStdOut.filter(el => !key.includes(el));
+  });
+
+  const stdErrs = [];
+  child.stderr.on('data', function (buf) {
+    const line = String(buf);
+    util.print('StdErr:', line);
+    if (allowedStdErr.filter(allowed => line.includes(allowed)).length === 0) {
+      stdErrs.push(line.trim());
+    }
+  });
+
+  const code = await new Promise(resolve => child.on('close', resolve));
+
+  const problems = [];
```

Comment:
would prefer if these were encapsulated a bit by functions so that the overall flow of this script can be easily grasped in a few lines.